pkg/reconcile/runner.go

```diff
@@ -0,0 +1,158 @@
+package reconcile
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"github.com/solo-io/autopilot/pkg/ezkube"
+	"github.com/solo-io/autopilot/pkg/utils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type Reconciler interface {
+	// reconcile an object
+	// requeue the object if returning an error, or a non-zero "requeue-after" duration
+	Reconcile(object ezkube.Object) (time.Duration, error)
```

Comment:
Do we also want to implement a "number of retries" concept? Or is that sort of pointless now that you are pulling the latest object?